### PR TITLE
[ASCII-1977] Use configuration pubsub system to update log_level

### PR DESCRIPTION
**Pull request by misteriaud**
### What does this PR do?

This PR moves the logger level update logic to a reactive pattern (using `config.OnUpdate()` registration function to react to configuration update).
It also moves `pkg/config/logs` package to `pkg/log/util/setup` for a better separation of concerns. 

### Describe how to test/QA your changes

QA done through unit-tests and manual tests.

### CODE REVIEW

File: pkg/util/log/setup/log.go [hunk 1]
```diff
@@ -117,6 +116,33 @@ func SetupLogger(loggerName LoggerName, logLevel, logFile, syslogURI string, sys
...
+	// Registering a callback in case of "log_level" update
+	cfg.OnUpdate(func(setting string, oldValue, newValue any) {
+		if setting != "log_level" || oldValue == newValue {
+			return
+		}
+		level := newValue.(string)
+
+		seelogLogLevel, err := log.ValidateLogLevel(level)
+		if err != nil {
+			return
```
Comment (remeh) [inline pkg/util/log/setup/log.go:129]: Since we can't return it here, would it be worth logging an errir in the various `if err != nil` of this func?

**Review by remeh — COMMENTED** [review]
Left a comment but otherwise LGTM.

File: pkg/config/aliases.go [hunk 2]
```diff
@@ -261,3 +261,15 @@ func LoadWithoutSecret() (*model.Warnings, error) {
 func GetProcessAPIAddressPort() (string, error) {
 	return pkgconfigsetup.GetProcessAPIAddressPort(Datadog())
 }
+
+// ChangeLogLevel changes the log level of the Datadog agent.
+// It takes a `level` string representing the desired log level and a `source` model.Source indicating the desired subconfiguration to update.
+// It returns an error if the log level is invalid or if there is an error setting the log level.
+func ChangeLogLevel(level string, source model.Source) error {
```
Comment (ogaca-dd) [inline pkg/config/aliases.go:268]: This file contains only aliases. This is used by OTEL. This function should probably go elsewhere.

File: pkg/util/log/setup/log.go (hunk 1 above)
Comment (misteriaud) [reply pkg/util/log/setup/log.go:130]: Good idea ! Done in 3bf2ae77434cc6e601b74f33ae125377a7f3ab89.

File: pkg/config/aliases.go (hunk 2 above)
Comment (misteriaud) [reply pkg/config/aliases.go]: Oh I didn't know, I moved it in its own file (I didn't found a relevant place to move it to). Done in 3bf2ae77434cc6e601b74f33ae125377a7f3ab89.

**Review by CelianR — APPROVED** [review]
Okay for devx-infra files 👍

**Review by ichinaski — APPROVED** [review]

**Review by FlorentClarret — APPROVED** [review]

**Review by chouetz — APPROVED** [review]

**Review by remeh — APPROVED** [review]
LGTM!

**Review by songy23 — APPROVED** [review]
LGTM for OTel

File: pkg/config/settings/runtime_setting_log_level.go [hunk 3]
```diff
@@ -52,16 +51,10 @@ func (l *LogLevelRuntimeSetting) Get(_ config.Component) (interface{}, error) {
...
 		return err
 	}
-
-	key := "log_level"
-	if l.ConfigKey != "" {
-		key = l.ConfigKey
-	}
-
-	config.Set(key, level, source)
+	config.Set("log_level", seelogLogLevel, source)
```
[the remark this answers is not in the document]
Comment (misteriaud) [reply pkg/config/settings/runtime_setting_log_level.go]: Done in 644d06c67ecfbb5cc7bfa160bb8aaf55f1edb989.

File: pkg/config/aliases.go [hunk 4]
```diff
@@ -261,3 +229,15 @@ func LoadWithoutSecret() (*model.Warnings, error) {
...
+	seelogLogLevel, err := log.ValidateLogLevel(level)
+	if err != nil {
+		return err
+	}
+	Datadog().Set("log_level", seelogLogLevel, source)
```
[the remark this answers is not in the document]
Comment (misteriaud) [reply pkg/config/aliases.go]: Done in 644d06c67ecfbb5cc7bfa160bb8aaf55f1edb989.

File: pkg/config/aliases.go (hunk 2 above)
Comment (misteriaud) [reply pkg/config/aliases.go]: Done in 644d06c67ecfbb5cc7bfa160bb8aaf55f1edb989.

**Review by hush-hush — APPROVED** [review]

**Review by purple4reina — APPROVED** [review]

**Pull request merged**